File: leptos/src/stores/process.rs
use leptos::set_interval_with_handle;
use leptos::{create_signal, on_cleanup, spawn_local, ReadSignal, Scope, SignalSet};

use crate::api::{get_process, Process};

pub fn process_store(cx: Scope, pid: u32) -> ReadSignal<Option<Process>> {
    let (process, set_process) = create_signal(cx, None);

    let handle = set_interval_with_handle(
        move || {
            spawn_local(async move {
                let proc = get_process(pid).await;
                set_process.set(proc);
            });
        },
        std::time::Duration::from_secs(1),
    )
    .unwrap();
    // FIXME: This cleanup does not appear to work
    on_cleanup(cx, move || {
        handle.clear();
    });

    process
}

File: solid/src/stores/process.ts
import { Store, createStore } from 'solid-js/store';
import { onCleanup } from 'solid-js';
import { getProcess } from "@/api";
import type { Process } from "@/api";

export interface Content {
    process?: Process;
}

export function processStore(pid: number): Store<Content> {
    const [process, setProcess] = createStore<Content>({});

    let interval = setInterval(() => {
        getProcess(pid).then((proc) => {
            setProcess('process', proc);
        });
    }, 1000);
    onCleanup(() => clearInterval(interval));

    return process;
}
